kong/plugins/zipkin/reporter.lua

```diff
@@ -4,19 +4,29 @@ local cjson = require "cjson".new()
 cjson.encode_number_precision(16)
 
 local floor = math.floor
-local gsub = string.gsub
 
 local zipkin_reporter_methods = {}
 local zipkin_reporter_mt = {
-  __name = "kong.plugins.zipkin.reporter",
   __index = zipkin_reporter_methods,
 }
 
+local localEndpoint = {
+  serviceName = "kong"
+}
+
+-- Utility function to set either ipv4 or ipv6 tags
+-- nginx apis don't have a flag to indicate whether an address is v4 or v6
+local function ip_kind(addr)
+  -- use the presence of "." to signal v4 (v6 uses ":")
+  if addr:find(".", 1, true) then
+    return "ipv4"
```

Comment:
[RFC 4191, §2.2.3](https://tools.ietf.org/html/rfc4291#section-2.2) specifies a valid use of dots in IPv6 addresses.  would be safer to check for colons (all IPv6 have at least two, no IPv4 does)

Reply:
changed, thanks!